src/task/blocking.rs

```diff
@@ -45,52 +133,175 @@ lazy_static! {
         let (sender, receiver) = bounded(0);
         Pool { sender, receiver }
     };
+
+    /// Sliding window for pool task frequency calculation
+    static ref FREQ_QUEUE: Mutex<VecDeque<u64>> = {
+        Mutex::new(VecDeque::with_capacity(FREQUENCY_QUEUE_SIZE.saturating_add(1)))
+    };
+
+    /// Dynamic pool thread count variable
+    static ref POOL_SIZE: Mutex<u64> = Mutex::new(LOW_WATERMARK);
 }
 
-// Create up to MAX_THREADS dynamic blocking task worker threads.
-// Dynamic threads will terminate themselves if they don't
-// receive any work after between one and ten seconds.
-fn maybe_create_another_blocking_thread() {
-    // We use a `Relaxed` atomic operation because
-    // it's just a heuristic, and would not lose correctness
-    // even if it's random.
-    let workers = DYNAMIC_THREAD_COUNT.load(Ordering::Relaxed);
-    if workers >= MAX_THREADS {
-        return;
+/// Exponentially Weighted Moving Average calculation
+///
+/// This allows us to find the EMA value.
+/// This value represents the trend of tasks mapped onto the thread pool.
+/// Calculation is following:
+/// ```text
+/// +--------+-----------------+----------------------------------+
+/// | Symbol |   Identifier    |           Explanation            |
+/// +--------+-----------------+----------------------------------+
+/// | α      | EMA_COEFFICIENT | smoothing factor between 0 and 1 |
+/// | Yt     | freq            | frequency sample at time t       |
+/// | St     | acc             | EMA at time t                    |
+/// +--------+-----------------+----------------------------------+
+/// ```
+/// Under these definitions formula is following:
+/// ```text
+/// EMA = α * [ Yt + (1 - α)*Yt-1 + ((1 - α)^2)*Yt-2 + ((1 - α)^3)*Yt-3 ... ] + St
+/// ```
+/// # Arguments
+///
+/// * `freq_queue` - Sliding window of frequency samples
+#[inline]
```

Comment:
in general, humans are bad at guessing whether something should be inlined. Usually in rust it's better to leave these attributes out unless come code is going to be executed in a super hot path and you've done work to ensure the improvement is measurable

Reply:
This comment is not contributing to the overall discussion + review. I’ve found it quite a bit `out of context` and also from the `book`. I've made the benchmarks already with and without and decided to leave it inlined. You can see below my last run again after your comment (run shouldn't be needed because of how compiler inlines and unrolls):

Mean of 10 consecutive runs of `blocking` benchmark:
```
43057207.9 ns without inlining
41816626.1 ns with inlining
```

That is also better for users of this library indirectly.

Reply:
I'm a bit skeptical that change is related to the inline here because this is a function that gets called 5 times per second, and that measurement skew is 2 milliseconds.